Makefile

```diff
@@ -1,3 +1,86 @@
+TEST_BATS_VERSION = v1.10.0
+TEST_BASH_VERSIONS = 3.2.57 4.1.17
```

Comment:
It would be preferable if we have some kind of moving targets for Bash versions so that we don't have to track Bash releases and make changes each time a new Bash release is made.

Reply:
Just add a new version in TEST_BASH_VERSIONS and all the needed target are "created"
All the targets are dynamic and based on those variables

Reply:
> Perhaps it's possible to somehow extract the versions used in other major distros?
Is is really practical to test with arbitrary Bash versions in Alpine?

You can add all the bash versions needed.
Testing different bash versions is only to be sure that code is portable to all bash version who could be used, so yes you could target only BASH_VERSION you like.
I don't find a gist I've once, but there is so many bugs in bash with array for example and another things

Reply:
Okay. Having an option to pick arbitrary versions is still great for a start! We can figure something out going forward if we wish to test in different distros with their stock and potentially patched versions.

Reply:
The version who could be added with this are only the ones available as official bash image on [dockerhub](https://hub.docker.com/_/bash)

I could check within few days, if you want, the possibility to use official distribution image published instead of the bash ones

Something like :

```
BASE_IMAGES: bash:3.2.57 ubuntu:22.0 alpine:10
```

Reply:
> I could check within few days, if you want, the possibility to use official distribution image published instead of the bash ones

Looking at the tags of the Bash image, they seem to indeed have rolling-release ones: e.g. `5` and `4` as well as `latest`.

For the practical goals above, It seems enough to test in `bash:latest` (or whatever is considered the latest stable version that we want to be ready for) and `bash:4` as well as `latest` and maybe 1 other non-EOL release of a few other mainstream distros.

This looks like a job for a separate PR -- so let's not hold up this one for that.